src/components/Header/H1.d.ts

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+declare namespace H1 {
+    export interface IH1Props<T> extends React.HTMLProps<T> {
+      className?: string;
+    }
+    export type H1Props = IH1Props<HTMLDivElement>;
```

Comment:
Glad to see you considered some of the suggestions.
I forgot to mention this but the `HTML*` type shouldn't necessarily always be a `HTMLDivElement`.
In most cases it is usually determined by the parent html element returned by the component.
However some html types such as headers `<h1>`, `<h2>`  and `<i>` haven't been implemented yet so for such you would consider using the generic type which is `HTMLElement`. 
You can find the `HTML*` types [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/global.d.ts)